test/convert_to_iso_string.test.ts

```diff
@@ -29,14 +30,14 @@ describe("convertToIsoString", () => {
   it("should convert moment objects to ISO strings in UTC", () => {
     const tzDate = "2023-01-01T01:00:00+01:00";
     const isoDate = "2023-01-01T00:00:00.000Z";
-    const converted = convertToIsoString(moment(tzDate));
+    const converted = convertToIsoString(moment(tzDate).toDate());
```

Comment:
The test is if it correctly converts moment objects, so should not be made into a date. The issue is with the typing in `convertToIsoString`

Reply:
But isn't there a bug here then?

Looking at `convertToIsoString`:

```typescript
export function convertToIsoString(data: string | Date) {
  if (
    data &&
    // if this is a date object of type moment or dayjs, or regular date object (all of them has toISOString)
    ((typeof data !== "string" && typeof data.toISOString === "function") ||
      // if it's a ISO string already
      (typeof data == "string" && isIsoDate(data)))
  ) {
    // wrap it new Date() to convert it to UTC based ISO string in case it is in another timezone
    try {
      return new Date(data).toISOString();
    } catch (err) {
      return null;
    }
  }

  return null;
}
```

If `data` can be a `Moment`, would the line `new Date(data)` then work?

Reply:
yep! 
![image](https://github.com/ftrackhq/ftrack-javascript/assets/2225828/86b70992-a5d8-4e64-856d-7f5cb72aff3a)